pages/language/ref/common.mdx

```diff
@@ -1,114 +1,266 @@
+
+Usage example:
+
+```tact
+ton("1");           // 10^9 nanoToncoins, which is equal to one Toncoin
+ton("0.1");         // 10^8 nanoToncoins, which is equal to 0.1 Toncoin
+ton("0.000000001"); // 1 nanoToncoin, which is equal to 10^-9 Toncoins
+```
 
 ## dump
+
 ```tact
-fun dump(arg: Any);
+fun dump(arg);
+```
+
+Prints the argument `arg` to the contract's debug console.
+
+Can be applied to special values ([`null{:tact}`](/book/optionals), `void`) and values of the following types:
+
+* [`Int{:tact}`][int]
+* [`Bool{:tact}`][bool]
+* [`Builder{:tact}`][p], [`Cell{:tact}`][p] or [`Slice{:tact}`][p]
+* [`String{:tact}`][p] or [`StringBuilder{:tact}`][p]
+* [`map<k, v>{:tact}`](/book/maps)
+
+Usage examples:
+
+```tact
+// Int
+dump(42);
+
+// Bool
+dump(true);
+dump(false);
+
+// Builder, Cell or Slice
+dump(beginCell());           // Builder
+dump(emptyCell());           // Cell
+dump(emptyCell().asSlice()); // Slice
+
+// String or StringBuilder
+dump("Hello, my name is..."); // String
+dump(beginTailString());      // StringBuilder
+
+// Maps
+let m: map<Int, Int> = emptyMap();
+m.set(2 + 2, 4);
+dump(m);
+
+// Special values
+dump(null);
+dump(emit("msg".asComment())); // As emit() function doesn't return a value, dump() would print #DEBUG#: void.
 ```
 
-Prints argument to the contract debug console. `debug` feature has to be enabled in `tact.config.json`.
+<Callout>
+
+  For this function to work, the compiler option `debug` has to be set to `true{:tact}` for the current project in the [`tact.config.json`](/book/config). Read more about debugging on the dedicated page: [Debugging](/book/debug).
+
+</Callout>
 
 ## context
+
 ```tact
 fun context(): Context;
 ```
-Return Struct `Context`, that consists of
 
-| Field   | Type    | Description                                                                                                               |
-|---------|---------|---------------------------------------------------------------------------------------------------------------------------|
-| bounced | Bool    | [Bounced](https://ton.org/docs/learn/overviews/addresses#bounceable-vs-non-bounceable-addresses) flag of incoming message |
-| sender  | Address | Address of sender                                                                                                         |
-| value   | Int     | Amount of nanoToncoins in message                                                                                         |
-| raw     | Slice   | Slice reminder of message
\ No newline at end of file
+Returns `Context{:tact}` [Struct](/book/structs-and-messages#structs), that consists of:
+
+Field   | Type                  | Description
+:------ | :-------------------- | :----------
+bounced | [`Bool{:tact}`][bool] | [Bounced](https://ton.org/docs/learn/overviews/addresses#bounceable-vs-non-bounceable-addresses) flag of the incoming message.
+sender  | [`Address{:tact}`][p] | Internal address of the sender on the TON blockchain.
```

Comment:
What do I use if I need to get the sender's address: the `sender()` function or `context().sender`?